src/components/ProjectCard/ProjectHoverCard.tsx

```diff
@@ -39,8 +41,25 @@ export const ProjectHoverCard: FC<ProjectCardProps> = ({
   const router = useRouter();
   const { data: POLPrice } = useFetchTokenPrice();
   const { data: activeRoundDetails } = useFetchActiveRoundDetails();
+  const { data: allProjects, isLoading } = useFetchAllProjects();
 
   const isQaccRoundEnded = useFetchMostRecentEndRound(activeRoundDetails);
+  const { data: allRounds } = useFetchAllRound();
+  const [isAllocationDone, setIsAllocationDone] = useState(false);
+
+  useEffect(() => {
+    const cheakAllProjectStatus = async () => {
+      if (allProjects && allRounds) {
+        const allProjectsReady = await checkAllProjectsStatus(
+          allProjects?.projects,
+          allRounds,
+        );
+        setIsAllocationDone(allProjectsReady);
+      }
+    };
+
+    cheakAllProjectStatus();
+  }, [allProjects, allRounds]);
```

Comment:
It's not efficient, it will load per project card. We can calculate it once in the `useFetchAllRound`

Reply:
the `useFetchAllRound`  only provides if the `isBatchMintingExecuted` or not ..we also need to check if the price of the tokens is up to date for the each project which is done here `src/services/tokenPrice.ts/getTokenPriceRangeStatus` that is why i used this

Reply:
We may find another place. The point is that logic is repeated twice in code, and in one of them (if I am not mistaken) will be executed in each card!
It's better move it to a hook with `useQuery` and cache it!